doc/proposal/monitoring.md

```diff
@@ -0,0 +1,144 @@
+  - [Mixer-Specific Telemetry](#Mixer-Specific_Telemetry)
+
+## Introduction
+
+The [Istio Service Mesh](http://github.com/istio/istio) provides automated 
+telemetry data collection for integrated services. As requests flow through the
+service mesh, request metadata is generated by the [Proxy](http://github.com/istio/proxy) 
+and sent to the [Mixer](http://github.com/istio/mixer). The Mixer repackages the
+received metadata, generating access logs and request metrics, and transmits the
+telemetry data to a configurable set of backend systems for monitoring and 
+alerting.
+
+Istio development has, so far, focused on providing the subsystems for the 
+reporting of telemetry data and for tailoring the collection to match the intent
+of the service operators. Little attention has been devoted to monitoring the 
+behavior of the individual Istio components (Proxy, [Manager](http://github.com/istio/manager),
+Mixer).
+
+This document proposes a high-level design for the monitoring of the Istio 
+Service Mesh itself.
+
+## Proposed Design
+
+The proposed design adopts two high-level goals:
+1. **Limit special handling for Istio components**
+   - Telemetry data for the Istio Proxy and Istio Manager components _should_ be
+     reported via the mixer through the same mechanisms as telemetry data for 
+     the non-Istio services within the service mesh.
+   - Telemetry data for the Istio Mixer itself _should_ be sent through the same 
+     processing code as data received via RPCs.
+1. **Prevent mixer-generated traffic from generating additional reports**
+   - In order to prevent a never-ending loop of reporting of metrics, traffic
+     generated by mixer to downstream services (e.g. logging/metrics backends)
+     _must_ be exempted from triggering the push of new telemetry data to 
+     backend systems.
+
+### Design Summary
+
+Each Istio component will be responsible for the following:
+1. Generating its own set of telemetry data (logs/metrics)
+1. Reporting that data to the Istio Mixer, on a regular basis, via the existing
+   Mixer API (see: [ReportRequest](https://github.com/istio/api/blob/master/mixer/v1/report.proto#L23)).
+1. Publishing the set of schemas for the telemetry data generated by its 
+   component in a *discoverable* fashion.
+
+In order to prevent a never-ending loop of Reports, Mixer will be configured to
+drop ReportRequests received where the source of the traffic being reported is
+Mixer itself.
+
+### Telemetry Generation and Reporting
+
+Schemas for telemetry data are necessary for Mixer to properly convert 
+ReportRequests into logs and metrics for various backend systems (see: 
+[LogEntryDescriptor](https://github.com/istio/api/blob/master/mixer/v1/config/descriptor/log_entry_descriptor.proto#L22),
+[MetricsDescriptor](https://github.com/istio/api/blob/master/mixer/v1/config/descriptor/metric_descriptor.proto#L53)).
+
+It would be extremely onerous to require that these schemas be kept in sync with
```

Comment:
I was considering all of the metrics that live in code that OP is not aware of, but are used for monitoring by internal systems. Most of the dashboards I've used for oncall monitoring have not been built from the metrics exposed through OP. Instead, we rely on an internal metrics library (think `/varz` here).

I was considering a dynamic setup, yes. Though, I'm not entirely convinced that this is optimal or even a good idea. It may be necessary, however (depending on how our config story evolves, etc.). The real issue with a dynamic setup, as I see it currently, is that Report doesn't take fully formed metric values. If it did, this would be much simpler. As it stands now, however, we need to export not just the definitions, but also how attributes map into those definitions, and then also ensure that the Reporting uses those exact attributes.